### PR TITLE
Add type declaration files support

### PR DESCRIPTION
**Pull request by roy2100**

### CODE REVIEW

**Comment by ArrayZoneYour** [comment]
missing `SM2.constants`

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      /** 
+       * Concatenation mode
+       * @default "C1C3C2"
+       *  */
+      mode?: "C1C3C2" | "C1C2C3";
+      /** 
+       * The encoding of the plain data string,if `data` is not a string then `inputEncoding` is ignored.
+       * @default "utf8"
+       */
+      inputEncoding?: string;
```
Comment (ArrayZoneYour) [inline types/index.d.ts]: `string` => `BufferEncoding`

**Review by ArrayZoneYour — CHANGES_REQUESTED** [review]

**Comment by leecade** [comment]
Plz add TS support description in Readme

**Comment by roy2100** [comment]
@ArrayZoneYour already add the type overloading, review please

**Review by ArrayZoneYour — APPROVED** [review]

**Pull request merged**